### ISSUE TITLE
BooleanField not working Bootstrap4

### ISSUE THREAD

**Issue by ahlan90** [django-crispy-forms/django-crispy-forms#787]
* Package version: 1.7.1
* Django version: 1.11.10
* Python version: 2.7
* Template pack: Bootstrap4

### Description:

Simple BooleanField() not render on my template with bootstrap4, but in "uni_form" working!

**Comment by jacopsd** [comment]
Was this fixed? Facing the same issue here, boolean field simply does not appear on the form with BS4

**Comment by ahlan90** [comment]
I used Material Pro Lite Template with Bootstrap4 and i fixed overwriting the css file of the template

**Comment by jacopsd** [comment]
So what was the fix if I may ask?

2018-06-01 15:54 GMT+02:00 Ahlan Guarnier <notifications@github.com>:

**Comment by carltongibson** [comment]
It looks like there's a bug in the template here. If we can pin that down exactly we could fix it.

**Comment by zoidbergwill** [comment]
Are you folks using latest master or the latest release? They seem to work on my test project.

Can you send me the HTML you're getting?

**Comment by jacopsd** [comment]
django-crispy-forms release 1.7.2, so latest indeed

Generated HTML:

```
<div class="form-group"> 
    <div id="div_id_new" class="checkbox"> 
        <label for="id_new" class=""> 
            <input type="checkbox" name="new" class="checkboxinput" id="id_new" checked />
                    New member
        </label>
        <p id="hint_id_new" class="help-block">For new members only</p> 
    </div>
</div>

```

On screen, only the text and help text is shown, no checkbox.

**Comment by carltongibson** [comment]
Hi @jacopsd. Thanks for posting the HTML. 

I'm afraid it leaves me more confused as to a possible cause here: that HTML is correct. Trying it out leads to the expected result. 

Can you open your web inspector, find the `input` element that's not appearing and look at the CSS or JS that's applied to it to see why it's not showing. (If it's there in the source and you disable styles and javascript it will appear, so something is overriding that.)

**Comment by jacopsd** [comment]
You're right.

Found it: my bootstrap4 style.css file says:

```
/* Remove default checkbox */
[type="checkbox"]:not(:checked),
[type="checkbox"]:checked {
  position: absolute;
  left: -9999px;
  opacity: 0; }
```

When I remove these lines, the issue is solved. I wonder why these lines are there...

**Comment by carltongibson** [comment]
The first example from the BS4 forms docs has this for the checkbox:

```
 <div class="form-group form-check">
    <input type="checkbox" class="form-check-input" id="exampleCheck1">
    <label class="form-check-label" for="exampleCheck1">Check me out</label>
  </div>
```

The classes are different. I’d guess that’s the source of the errror. 

Fancy adjusting the template to see if it works and opening a PR is so?

**Comment by ahlan90** [comment]
$(document).ready(function() {
         
      $( ":checkbox" ).each(function( index ) {
          $(this).prependTo($("#div_" + $(this).attr('id')))
        });
      
     });

**Comment by goromachine** [comment]
Hi, I'm issuing this problem too, and the JS that @ahlan90 proposed didn't worked for me. There is any way to have this fixed on the next version?
I know I could do a PR for that, but I don't know how to solve it. 

Thank you

**Comment by LiamK** [comment]
I just ran into this to.

**Comment by voidus** [comment]
I am trying to replicate this issue but don't really now the conditions. If anybody could show a forms.py and template or even better create a django project that has this issue, I'll try to fix it.

**Comment by GaganChhabra88** [comment]
I have also faced this and fortunately was able to fix this for myself. In this scenario, Django is rendering labels with css 'disply:none' so the checkboxes, radio selects and dropdown choices are not getting displayed.

I wrote a simple javascript to fix this in my template:

var elems = document.querySelectorAll('label.form-check-label');
        for (i = 0; i < elems.length; i++) {
            elems[i].style.display = "inline";
        }


Replication steps are easy:

Settings:
     CRISPY_TEMPLATE_PACK = 'bootstrap4'
1. Include a BooleanField in your Model as below:
      acceptance_check = models.BooleanField(default=True)
2. Include this in forms as below:
      acceptance_check = forms.BooleanField(required=True) 
3. Render your form in your template.
You will not be able to see BooleanField(checkbox) on screen.
